src/main/scala/tech/cryptonomic/conseil/config/Security.scala

```diff
@@ -1,8 +1,47 @@
 package tech.cryptonomic.conseil.config
 
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import cats.effect.IO
+import cats.effect.concurrent.Ref
+import com.typesafe.scalalogging.LazyLogging
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport
 import pureconfig.generic.auto._
 
-object Security {
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object Security extends ErrorAccumulatingCirceSupport with LazyLogging {
+
+  /** Keys cache */
+  private val nautilusCloudKeys: Ref[IO, Set[String]] = Ref.of[IO, Set[String]](Set.empty).unsafeRunSync()
+
+  /** Updates API keys from Nautilus-Cloud endpoint */
+  def updateKeys(
+      ncc: NautilusCloudConfiguration
+  )(implicit executionContext: ExecutionContext, system: ActorSystem, mat: Materializer): Unit = {
+    val update = for {
+      apiKeys <- Http()
+        .singleRequest(
+          HttpRequest(uri = makeUri(ncc))
```

Comment:
Can this handle HTTPS requests?

Reply:
yes, the same method is used in `TezosNodeOperator` which usually uses HTTPS for requests